ui/plugins/controller.js

```diff
@@ -278,37 +278,50 @@ treeherder.controller('PluginCtrl', [
-                    // second is when we have the buildapi id and need to send a request
-                    // to the self serve api (which does not listen over pulse!).
-                    ThJobModel.retrigger($scope.repoName, $scope.job.id).then(function() {
-                        // XXX: Bug 1170839 disables buildapi retrigger requests for the ash branch
-                        if($scope.repoName === "ash") {
-                            return;
-                        }
-                        // XXX: Remove this after 1134929 is resolved.
-                        var requestId = getBuildbotRequestId();
-                        if (requestId) {
-                            return thBuildApi.retriggerJob($scope.repoName, requestId);
-                        }
-                    }).then(function() {
-                        thNotify.send("Retriggered job: " + $scope.jobSearchStr,
-                                      'success');
-                    }).catch(function(e) {
-                        // Generic error eg. the user doesn't have LDAP access
-                        thNotify.send(
-                            ThModelErrors.format(e, "Unable to send retrigger"), 'danger');
-                    });
-                } else {
-                    thNotify.send("Job not yet loaded for retrigger", 'warning');
+                var job_ids = [];
+                for(var i=0; i<jobs.length; i++){
+                    // Only enter a retrigger if we have a valid loaded job, if the user
+                    // tries to retrigger eg. via shortcut before the load we warn them
+                    if (jobs[i].id) {
+                        job_ids.push(jobs[i].id);
+                    } else {
+                        thNotify.send("Job not yet loaded for retrigger", 'warning');
+                    }
                 }
+                // The logic here is somewhat complicated because we need to support
+                // two use cases the first is the case where we notify a system
+                // other then buildbot that a retrigger has been requested. The
+                // second is when we have the buildapi id and need to send a request
+                // to the self serve api (which does not listen over pulse!).
+                ThJobModel.retrigger($scope.repoName, job_ids).then(function() {
+                    // XXX: Bug 1170839 disables buildapi retrigger requests for the ash branch
+                    if($scope.repoName === "ash") {
+                        return;
+                    }
+                    // XXX: Remove this after 1134929 is resolved.
+                    // Initial promise for sequential execution.
+                    var jobPromise = Promise.resolve();
+                    jobs.forEach(function(job) {
+                        jobPromise = jobPromise.then(function() {
+                            return selectJob(job.id);
```

Comment:
It should return a list of artifacts with name `buildapi` belonging to the jobs listed in job_id_list. I forgot to mention that job_id_list must be a comma separated list of job ids, so in the code above
```js
var job_id_list = _.pluck(jobs, 'id').join(',')
```
The resulting url should be something like
https://treeherder.mozilla.org/api/project/mozilla-inbound/artifact/?name=buildapi&type=json&job_id__in=12475518,12475512,12475511